src/SimpleAuth.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+/// @title SimpleAuth
+/// @author Anna Carroll <https://github.com/anna-carroll/3074>
+abstract contract SimpleAuth {
```

Comment:
pedantic nit: maybe `BaseAuth` instead?

This can be sort of the "unsafe" interface, where `Auth` is a higher-level interface that handles errors, bubbles up rd, etc.

Reply:
nice, agree https://github.com/clabby/eip-3074-foundry/pull/8/commits/f33d7d11506159f759ab170bf40b605833a305fb